pricegraph/src/api/price_estimation.rs

```diff
@@ -122,6 +122,10 @@ impl Pricegraph {
             buy: pair.sell,
             sell: pair.buy,
         };
+
+        if limit_price < 1e-15 {
+            return None;
+        }
         let max_xrate = LimitPrice::new(limit_price)?.exchange_rate().inverse();
```

Comment:
TODO: How do we properly solve this? The problem is that a very low limit price

(the fuzzer found
```
`0.000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000059755401472367
```
)

causes the inverse to become infinite which fails the debug assert without this check.

Since this is an extreme edge cache I am okay with forcing the fuzzer to have more reasonable limit prices but I am not sure whether it is better to have some kind of check here instead. I chose `1e-15`  just to illustrate the point.

Reply:
We should create an issue. I think this is definitely the case of an incorrect assumption about floating point math in the `pricegraph` that we should fix.